fuzz-test-suite/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# Determine the flags for fuzzing. Use OSS-Fuzz's configuration if available, otherwise fall back to defaults.
+if(DEFINED ENV{LIB_FUZZING_ENGINE})
+    set(FUZZING_ENGINE $ENV{LIB_FUZZING_ENGINE})
+    set(FUZZING_COMPILE_FLAGS "")
+    set(FUZZING_LINK_FLAGS "${FUZZING_ENGINE}")
+else()
+    set(FUZZING_COMPILE_FLAGS "-fsanitize=fuzzer")
+    set(FUZZING_LINK_FLAGS "-fsanitize=fuzzer")
+endif()
+
+# Define the fuzz target
+add_executable(DateParserFuzzer dateparserfuzzer.cpp)
```

Comment:
If more fuzzer test cases will be added in the future, then perhaps the target can be renamed to something more generic like `ql_fuzz_test_suite` like the existing [test suite target](https://github.com/lballabio/QuantLib/blob/master/test-suite/CMakeLists.txt#L389-L390).

Reply:
Cool I'll take a look, I admit I haven't extended cmake in this way before, but I'll take a swing at it.

Reply:
After looking into this a little I don't think that this makes sense. The boost harnesses that you are referencing here can be bundled because each unit-test has a finite runtime. So each test can be either serialized or run in parallel from a single executable. In contrast, a fuzz-test will run;
- Indefinitely.
- Or until a crashing input is found.
- Or you manually exit.
- Or if you manually set a timeout.

This poses a problem when you are bundling testing harnesses, as you can't neccesarily say that a fuzz harness is going to complete/exit or not. So if you bundle 3 harnesses, you might just get stuck on harness 1, resulting in 2 and 3 not getting fuzzed at all. So you typically want a single executable per functionality/functionalilty group.